consensus/qbft/core/preprepare.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+import (
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/consensus"
+	"github.com/ethereum/go-ethereum/consensus/qbft/message"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+func (c *core) sendPreprepareMsg(request *Request) {
+	logger := c.logger.New("state", c.state)
+
+	// If I'm the proposer and I have the same sequence with the proposal
+	if c.current.Sequence().Cmp(request.Proposal.Number()) == 0 && c.IsProposer() {
+		curView := c.currentView()
+		preprepare := message.NewPreprepare(curView.Sequence, curView.Round, request.Proposal)
+		preprepare.SetSource(c.Address())
+
+		// Sign payload
+		encodedPayload, err := preprepare.EncodePayloadForSigning()
+		if err != nil {
+			logger.Error("QBFT: Failed to encode payload of pre-prepare message", "msg", preprepare, "err", err)
```

Comment:
for consistency between log message we should decide whether to always use a message name which is lower case or upper-case.

`pre-prepare` or `PRE-PREPARE`